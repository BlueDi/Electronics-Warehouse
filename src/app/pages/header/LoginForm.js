import React, { Component } from 'react';
import { withCookies } from 'react-cookie';
import { withRouter } from 'react-router-dom';
import { Form, Header, Image, Message } from 'semantic-ui-react';
import { service } from '@utils';

class LoginForm extends Component {
  state = { error: false, name: '', password: '' };

  handleChange = (e, { name, value }) =>
    this.setState({ error: false, [name]: value });

  handleAuthenticate = () => {
    service
      .post('/login', this.state)
      .then(response => {
        this.setupLoggedUser(response.data);
      })
      .catch(() => {
        this.failedLogin();
      });
  };

  setupLoggedUser = value => {
    const { cookies, history } = this.props;
    for (var property in value) {
      cookies.set(property, value[property], { path: '/' });
    }
    cookies.set('cart', [], { path: '/' });
    history.push(value.userPath);
  };

  failedLogin = () => {
    this.setState({ error: true });
  };

  render() {
    return (
      <React.Fragment>
        <Header size="large">
          <Image src="/logo.png" /> Login to your account
        </Header>
        <Form
          size="large"
          error={this.state.error}
          onSubmit={this.handleAuthenticate}
        >
          <Form.Input
            fluid
            icon="user"
            iconPosition="left"
            name="name"
            placeholder="E-mail address"
            required
            onChange={this.handleChange}
          />
          <Form.Input
            fluid
            icon="lock"
            iconPosition="left"
            name="password"
            placeholder="Password"
            required
            type="password"
            onChange={this.handleChange}
          />
          <Message
            error
            header="Failed Login"
            content="Please check again your user's info."
          />
          <Form.Button fluid>Login</Form.Button>
        </Form>
      </React.Fragment>
    );
  }
}

export default withRouter(withCookies(LoginForm));
