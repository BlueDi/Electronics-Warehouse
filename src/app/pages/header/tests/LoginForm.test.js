import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import LoginForm from '@pages/header/LoginForm';

describe('<LoginForm />', () => {
  test('Renders properly', () => {
    const component = renderer.create(
      <Router>
        <LoginForm />
      </Router>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
