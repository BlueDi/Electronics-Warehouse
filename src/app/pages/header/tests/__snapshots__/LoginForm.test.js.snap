// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> Renders error message properly 1`] = `
Array [
  <div
    className="ui large header"
  >
    <img
      className="ui image"
      src="test-file-stub"
    />
     Login to your account
  </div>,
  <form
    className="ui large form"
    onSubmit={[Function]}
  >
    <div
      className="required field"
    >
      <label>
        Name or E-mail
      </label>
      <div
        className="ui fluid left icon input"
      >
        <input
          name="name"
          onChange={[Function]}
          placeholder="Name or E-mail"
          required={true}
          type="text"
        />
        <i
          aria-hidden="true"
          className="user icon"
        />
      </div>
    </div>
    <div
      className="required field"
    >
      <label>
        Password
      </label>
      <div
        className="ui fluid left icon input"
      >
        <input
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
        <i
          aria-hidden="true"
          className="lock icon"
        />
      </div>
    </div>
    <div
      className="ui error message"
    >
      <div
        className="content"
      >
        <div
          className="header"
        >
          Failed Login
        </div>
        <p>
          
        </p>
      </div>
    </div>
    <div
      className="field"
    >
      <button
        className="ui fluid button"
        onClick={[Function]}
      >
        Login
      </button>
    </div>
  </form>,
]
`;

exports[`<LoginForm /> Renders properly 1`] = `
Array [
  <div
    className="ui large header"
  >
    <img
      className="ui image"
      src="test-file-stub"
    />
     Login to your account
  </div>,
  <form
    className="ui large form"
    onSubmit={[Function]}
  >
    <div
      className="required field"
    >
      <label>
        Name or E-mail
      </label>
      <div
        className="ui fluid left icon input"
      >
        <input
          name="name"
          onChange={[Function]}
          placeholder="Name or E-mail"
          required={true}
          type="text"
        />
        <i
          aria-hidden="true"
          className="user icon"
        />
      </div>
    </div>
    <div
      className="required field"
    >
      <label>
        Password
      </label>
      <div
        className="ui fluid left icon input"
      >
        <input
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
        <i
          aria-hidden="true"
          className="lock icon"
        />
      </div>
    </div>
    <div
      className="ui error message"
    >
      <div
        className="content"
      >
        <div
          className="header"
        >
          Failed Login
        </div>
        <p>
          
        </p>
      </div>
    </div>
    <div
      className="field"
    >
      <button
        className="ui fluid button"
        onClick={[Function]}
      >
        Login
      </button>
    </div>
  </form>,
]
`;
